.github/workflows/ci.yml

```diff
@@ -67,14 +73,21 @@ jobs:
 
       - name: Install Python packages
         run: |
-          pip install --upgrade codecov tox setuptools
```

Comment:
Why remove setuptools for this list?

Reply:
I incidentally found that installing setuptools in this step is not necessary to run tox in the next step.  It wasn't really doing anything here.

Reply:
Let's revert this change.  `pip install --upgrade ...` is inherently unstable over time.  The idea of this line is that on *some* runs you might need a newer setuptools than is pre-installed in order for the rest of things to work.

This is a heuristic, of course.  It is equally true that on some runs you might need an *older* setuptools than `pip install --upgrade` is going to give you for the rest of things to work.

Changing this heuristic seems like it is out of scope for this PR, though.